src/disjoint_set.jl

```diff
@@ -129,17 +133,18 @@ num_groups(s::DisjointSets) = num_groups(s.internal)
 
 Finds the root element of the subset in `s` which has the element `x` as a member.
 """
-find_root{T}(s::DisjointSets{T}, x::T) = find_root(s.internal, s.intmap[x])
+find_root{T}(s::DisjointSets{T}, x::T) = s.revmap[find_root(s.internal, s.intmap[x])]
 
 in_same_set{T}(s::DisjointSets{T}, x::T, y::T) = in_same_set(s.internal, s.intmap[x], s.intmap[y])
 
 for f in (:union!, :root_union!)
     @eval begin
-        ($f){T}(s::DisjointSets{T}, x::T, y::T) = ($f)(s.internal, s.intmap[x], s.intmap[y])
+        ($f){T}(s::DisjointSets{T}, x::T, y::T) = s.revmap[($f)(s.internal, s.intmap[x], s.intmap[y])]
     end
 end
```

Comment:
I realize that the for loop for this definition was there before, but since there are only two functions and one definition inside the loop per function, I think it would be clearer if both functions were simply written out.